feat.: Azure Docs (#20706)
newrelic/docs-website: src/content/docs/serverless-function-monitoring/azure-function-monitoring/linux.mdx

    2. Use your preferred publishing mechanism to deploy your updated application to Azure. 
    
    
    When you deploy your application, the .NET agent is installed in the `/home/site/wwwroot/newrelic` folder.
    
    </TabsPageItem>

  </TabsPages>
</Tabs>

</Step>

<Step>

## Configure environment variables

After you've published the Azure Function app, [configure environment variables](/docs/serverless-function-monitoring/azure-function-monitoring/env-variables-azure):

  1. Navigate to your Azure Function in the Azure portal.
  2. Under **Settings**, click **Environment variables**, and then click **Advanced edit**.
  3. Based on your deployment runtime, paste the following values:

<Callout variant="important">
Ensure that you add a comma at the end of the last existing line and update your license key in the following configurations.
</Callout>

<Tabs>
  <TabsBar>
    
    <TabsBarItem id="nodejs">Node.js</TabsBarItem>
    <TabsBarItem id="py">Python</TabsBarItem>
    <TabsBarItem id=".net">.NET</TabsBarItem>

  </TabsBar>

  <TabsPages>

  <TabsPageItem id="nodejs">

  ```json
  {

>     Information required for Node.js agent

Reviewer: "NODE_OPTIONS": "-r newrelic",
"NEW_RELIC_LICENSE_KEY": "licenseKey",
"NEW_RELIC_APP_NAME": "Name of Function App"

>     Information required for Node.js agent

Reviewer: ```suggestion
   "NODE_OPTIONS": "-r newrelic",
   "NEW_RELIC_LICENSE_KEY": "ingest license key",
   "NEW_RELIC_APP_NAME": "Name of Function App"
```

    <TabsBarItem id="python">Python</TabsBarItem>
    <TabsBarItem id="net">.NET</TabsBarItem>

  </TabsBar>

  <TabsPages>

    <TabsPageItem id="node">
    
    1. In the root directory of your Node.js Azure Function App project, install the New Relic agent as a project dependency by running:
    
      ```node
      npm i newrelic
      ```
    2. To ensure that the New Relic agent is installed with your project dependencies during the Azure deployment process, set the `SCM_DO_BUILD_DURING_DEPLOYMENT` to `true` in your application settings by running:

        ```node
        az functionapp config appsettings set --name {NAME_OF_YOUR_AZURE_FUNCTION_APP} --settings SCM_DO_BUILD_DURING_DEPLOYMENT=true
        ```
    
    3. Publish your Azure Function App using the Azure CLI or your preferred publishing method. You can use one of the following command to publish your Azure Function App:

        ```node
        func azure functionapp publish {NAME_OF_YOUR_AZURE_FUNCTION_APP}
        ```
        OR

        ```node
        az functionapp app up
        ```
        
        OR
        
        ```node
        az functionapp deploy
        ```

      <Callout variant="important">

>         The above provided commands are examples and may require additional parameters based on your specific Azure configuration and deployment strategy such as `OS`, `Node.js version`, `Azure resource group`, `Azure subscription`, `Azure storage account`, `Azure region`, and so on. Refer to Azure documentation to know more about these parameters for [`func azure functionapp publish`](https://learn.microsoft.com/en-us/azure/azure-functions/functions-core-tools-reference?tabs=v2#func-azure-functionapp-publish), [`az functionapp up`](https://learn.microsoft.com/en-us/cli/azure/functionapp/app?view=azure-cli-latest#az-functionapp-app-up), and [`az functionapp deploy`](https://learn.microsoft.com/en-us/cli/azure/functionapp?view=azure-cli-latest#az-functionapp-deploy) methods.

Reviewer 2: ```suggestion
        The commands shown here are examples and may require additional parameters based on your specific Azure configuration and deployment strategy such as `OS`, `Node.js version`, `Azure resource group`, `Azure subscription`, `Azure storage account`, `Azure region`, and so on. Refer to Azure documentation to know more about these parameters for [`func azure functionapp publish`](https://learn.microsoft.com/en-us/azure/azure-functions/functions-core-tools-reference?tabs=v2#func-azure-functionapp-publish), [`az functionapp up`](https://learn.microsoft.com/en-us/cli/azure/functionapp/app?view=azure-cli-latest#az-functionapp-app-up), and [`az functionapp deploy`](https://learn.microsoft.com/en-us/cli/azure/functionapp?view=azure-cli-latest#az-functionapp-deploy) methods.
```

      </Callout>

    </TabsPageItem>

    <TabsPageItem id="python">
    
    1. In the root directory of your Python Azure Function App project, open the `requirements.txt` file.
    2. Add the following line to your `requirements.txt` file to install the New Relic agent for Azure Function instrumentation:
        
        ```bash

>           git+https://github.com/newrelic/newrelic-python-agent.git@azure-function-instrumentation

Reviewer 3: ```suggestion
          newrelic
```

      <CollapserGroup>
        <Collapser
          id="option1"
          title="Option 1: Manually update the function_app.py file"
        >
          1. Open the your `function_app.py` file in your project.
          2. To import the New Relic agent, add the following statement to the top of the file:
          
            ```python
            import newrelic.agent
            ```
          3. Initialize the New Relic agent. For more information, refer to [Initialize Python agent.](/docs/apm/agents/python-agent/python-agent-api/initialize-python-agent-api/)
          4. Register the New Relic agent with your Azure Function. For more information, refer to [Register Python agent.](/docs/apm/agents/python-agent/python-agent-api/registerapplication-python-agent-api/)

            <Callout variant="important">
              The New Relic agent uses environment variables to name your application in the New Relic UI. Ensure that either `NEW_RELIC_APP_NAME` (recommended) or `WEBSITE_SITE_NAME` (Azure's default site name variable) is available as an environment variable for your Function App.
            </Callout>

>             Your `function_app.py` file should resembles the following structure:

Reviewer 2: ```suggestion
            Your `function_app.py` file should resemble the following structure:
```

                #
                #####################################################################

                mkdir original
                mv function_app.py ./original/

                echo -e 'import newrelic.agent\nimport os\n\nnewrelic.agent.initialize()\napp_name = os.environ.get("NEW_RELIC_APP_NAME", os.environ.get("WEBSITE_SITE_NAME", None))\nnewrelic.agent.register_application(app_name)\n' | cat - ./original/function_app.py > function_app.py

                echo 'original' >> .funcignore
              
              ```

            3. Save the `initialization.sh` file.
            4. Provide the permissions to script file by running:
            
              ```bash
              chmod 755 initialization.sh
              ```
            5. Run the script from the root directory of your project:
            
              ```bash
              ./initialization.sh
              ```
            </Collapser>

        </CollapserGroup>

    4. Publish your Azure Function App using the Azure CLI or your preferred publishing method. You can use the following command to publish your Azure Function App:
    
        ```bash

>         func azure functionapp publish \ ${NAME_OF_YOUR_AZURE_FUNCTION_APP}

Reviewer 3: ```suggestion
        func azure functionapp publish ${NAME_OF_YOUR_AZURE_FUNCTION_APP}
```

        ```

    </TabsPageItem>

    <TabsPageItem id="net">

    1. Add the latest version of the `NewRelic.Agent` NuGet package to your application project.
    2. Use your preferred publishing mechanism to deploy your updated application to Azure. 
    
    
    When you deploy your application, the .NET agent is installed in the `/home/site/wwwroot/newrelic` folder.
    
    </TabsPageItem>

  </TabsPages>
</Tabs>

</Step>

<Step>

## Configure environment variables

After you've published the Azure Function app, [configure environment variables](/docs/serverless-function-monitoring/azure-function-monitoring/env-variables-azure):

  1. Navigate to your Azure Function in the Azure portal.
  2. Under **Settings**, click **Environment variables**, and then click **Advanced edit**.
  3. Based on your deployment runtime, paste the following values:

<Callout variant="important">
Ensure that you add a comma at the end of the last existing line and update your license key in the following configurations.
</Callout>

<Tabs>
  <TabsBar>
    
    <TabsBarItem id="nodejs">Node.js</TabsBarItem>

>     <TabsBarItem id="py">Python</TabsBarItem>

Reviewer 3: ```suggestion
    <TabsBarItem id="python">Python</TabsBarItem>
```

    <TabsBarItem id=".net">.NET</TabsBarItem>

  </TabsBar>

  <TabsPages>

  <TabsPageItem id="nodejs">

  ```json
  {
    "NODE_OPTIONS": "-r newrelic",
    "NEW_RELIC_LICENSE_KEY": "YOUR-NEW-RELIC-INGEST-LICENSE-KEY",
    "NEW_RELIC_APP_NAME": "NAME-OF-THE-FUNCTION-APP"
  }
  ```

  </TabsPageItem>

>   <TabsPageItem id="py">

Reviewer 3: ```suggestion
  <TabsPageItem id="python">
```

  ```json
  {

>    Information required for Python agent

Reviewer 3: ```suggestion
    "NEW_RELIC_LICENSE_KEY": "YOUR-NEW-RELIC-LICENSE-KEY",
    "NEW_RELIC_APP_NAME":  "NAME-OF-THE-FUNCTION-APP",
    "PYTHONPATH": "${PYTHONPATH}:/home/site/wwwroot:/home/site/wwwroot/.python_packages/lib/site-packages",
    "FUNCTIONS_WORKER_RUNTIME": "python",
    "PYTHON_ENABLE_WORKER_EXTENSIONS": 1
```
